tests/probes/yamlfilecontent/types.yaml

```diff
@@ -0,0 +1,20 @@
+bool_true_1: true
+bool_true_2: True
+bool_true_3: TRUE
+bool_false_1: false
+bool_false_2: False
+bool_false_3: FALSE
+int_10: 42
+int_10_neg: -17
+int_8: 0o33
+int_16: 0xFF
+float: 7.4
+float_neg: -0.3
+float_exp: +12e03
+float_exp_neg: -43e-4
+float_inf: .inf
+float_nan: .NAN
+bool_true_cast: !!bool "true"
```

Comment:
How about adding tests to exercise the error path? i.e:
```C
	} else if (!question) {
		return NULL;
	}
```
The content would be something like:
```yaml
bool_error_cast: !!bool  "truu"
int_error_cast: !!int "50%"
```